src/Surfnet/ServiceProviderDashboard/Application/Metadata/OauthClientCredentialsClientJsonGenerator.php

```diff
@@ -0,0 +1,380 @@
+
+        $metadata += $this->generateOidcClient($entity);
+
+        if ($entity->getMetaData()->getLogo() !== null && $entity->getMetaData()->getLogo()->getUrl() !== '') {
+            $metadata += $this->generateLogoMetadata($entity);
+        }
+
+        return $metadata;
+    }
+
+    /**
+     * @param ManageEntity $entity
+     * @return array
+     */
+    private function generateOidcClient(ManageEntity $entity)
+    {
+        $metadata = [];
+        if ($entity->getOidcClient()) {
+            $secret = $entity->getOidcClient()->getClientSecret();
+            if ($secret) {
+                $metadata['secret'] = $secret;
+            }
+            // Reset the redirect URI list in order to get a correct JSON formatting (See #163646662)
+            $metadata['redirectUrls'] = $entity->getOidcClient()->getRedirectUris();
+            $metadata['grants'] = $entity->getOidcClient()->getGrants();
+            $metadata['accessTokenValidity'] = $entity->getOidcClient()->getAccessTokenValidity();
+            $metadata['isPublicClient'] = $entity->getOidcClient()->isPublicClient();
+        }
+        return $metadata;
+    }
+
+    private function generateAllContactsMetadata(ManageEntity $entity): array
+    {
+        $metadata = [];
+        $index = 0;
+
+        $contacts = $entity->getMetaData()->getContacts();
+
+        if ($contacts) {
+            if ($contacts->findSupportContact()) {
+                $metadata += $this->generateContactMetadata(
+                    'support',
+                    $index++,
+                    $contacts->findSupportContact()
+                );
+            }
+
+            if ($contacts->findAdministrativeContact()) {
+                $metadata += $this->generateContactMetadata(
+                    'administrative',
+                    $index++,
+                    $contacts->findAdministrativeContact()
+                );
+            }
+
+            if ($contacts->findTechnicalContact()) {
+                $metadata += $this->generateContactMetadata(
+                    'technical',
+                    $index++,
+                    $contacts->findTechnicalContact()
+                );
+            }
+        }
+        return $metadata;
+    }
+
+    private function generateOrganizationMetadata(ManageEntity $entity): array
+    {
+        $metadata = [];
+        $organization = $entity->getMetaData()->getOrganization();
+        if ($organization) {
+            $metadata = [
+                'OrganizationName:en' => $organization->getNameEn(),
+                'OrganizationName:nl' => $organization->getNameNl(),
+            ];
+        }
+        return array_filter($metadata);
+    }
+
+    private function generateContactMetadata($contactType, $index, Contact $contact): array
+    {
+        $metadata = [
+            sprintf('contacts:%d:contactType', $index) => $contactType,
+        ];
+
+        if (!empty($contact->getGivenName())) {
```

Comment:
All these if statements can be abstracted into a method, they always follow the exact same pattern.

Reply:
I agree, but the method would be rather messy in my opinion. Using the accessors on the contacts would require using something like $contact->$contactGetterName(). That kinda sucks.

Reply:
hm, a matter of preference whether it's messy or not (i find it a beautiful php feature).  If you dislike it though, it makes little sense to refactor.  I can live with how it is atm.